javacutil/src/main/java/org/checkerframework/javacutil/TypesUtils.java

```diff
@@ -333,7 +328,8 @@ public static boolean isString(TypeMirror type) {
      * @return true iff type represents a boolean type
      */
     public static boolean isBooleanType(TypeMirror type) {
-        return isDeclaredOfName(type, "java.lang.Boolean") || type.getKind() == TypeKind.BOOLEAN;
+        TypeKind kind = TypeKindUtils.primitiveOrBoxedToTypeKind(type);
+        return kind == TypeKind.BOOLEAN;
```

Comment:
```suggestion
        return type.getKind() == TypeKind.BOOLEAN || isDeclaredOfName(type, "java.lang.Boolean");
```

Reply:
I would probably keep the special handling for these. I would switch the order of the checks, to hopefully get the quick enum comparison first.

Reply:
Ok, I have kept the special handling.